Assets/Plugins/Source/Core/RuntimeConfig.cs

```diff
@@ -0,0 +1,259 @@
+/*
+ * Copyright (c) 2024 PlayEveryWare
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+//#define EOS_RUNTIME_NEW_CONFIG_SYSTEM
+
+namespace PlayEveryWare.EpicOnlineServices
+{
+    using Epic.OnlineServices.Auth;
+    using Epic.OnlineServices.IntegratedPlatform;
+    using Epic.OnlineServices.Platform;
+    using System;
+
+#if EOS_RUNTIME_NEW_CONFIG_SYSTEM
+    /// <summary>
+    /// Contains the information necessary to configure the EOS SDK to connect
+    /// properly to Epic Online Services. Is ignorant of platform, because it
+    /// _only_ represents the _current_ runtime values to use. The values of the
+    /// field members defined within will need to be determined at runtime.
+    ///
+    /// _Project Settings_ is the area where there will be options to define
+    /// configuration values on a platform by platform basis. The EOS Plugin
+    /// allows users who are making their game with EOS to define different
+    /// properties for each platform, and to store sandbox / deployment
+    /// overrides so that it is easy to switch between deployment environments
+    /// during development. Those settings are distinct from these set of
+    /// values.
+    ///
+    /// Most of the values for these field members come from the
+    /// [Development Portal](https://dev.epicgames.com/portal/)
+    /// </summary>
+    public readonly struct RuntimeConfig
```

Comment:
There are two kinds of 'runtime' configuration, bootstrap config values and then 'everything else'. If possible, I would prefer the 'everything else' not to be mixed with the bootstrap configuration values. The reason they were mixed together in the first place is because 1: it was faster 2: it allowed per platform overrides because it could hitch along with the existing override system.

Reply:
Which of these values pertain to bootstrap configuration?

Reply:
It's a little bit easier to answer it the other way around: 

These values are _not_ boostrap configuration, i.e. they're not needed for the EOS SDK bootstrap.
```
    public readonly bool AlwaysSendInputToOverlay;
    public readonly float InitialButtonDelayForOverlay;
    public readonly float RepeatButtonDelayForOverlay;
    public readonly bool SendInputDirectlyToSDK;
```

Reply:
Got it. Just to clarify - these ARE currently defined within EOSConfig, so you are suggesting that we redefine these to not be set at runtime? That change seems out of scope with the intent of the PR.

Reply:
Per conversation "offline" and in person, the definition of bootstrap in this context refers to the configuration values that are utilized during the setup / initialization of the plugin, not settings that are _required_ components of making the EOS plugin work.

Reply:
I removed these values from `RuntimeConfig`.